src/main/java/com/microsoft/util/CommentHelper.java

```diff
@@ -0,0 +1,89 @@
+package com.microsoft.util;
+
+import com.sun.source.doctree.DocTree;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import javax.lang.model.element.Element;
+
+public class CommentHelper {
+    public Element element;
+    public List<? extends DocTree> inlineTags = Collections.emptyList();
+    private Utils utils;
+    private boolean hasInheritDocTag = false;
+
+    public CommentHelper(Element element, Utils utils) {
+        this.element = element;
+        this.utils = utils;
+        this.inlineTags = utils.getFullBody(element);
+        this.hasInheritDocTag = utils.hasInlineTag(inlineTags, DocTree.Kind.INHERIT_DOC);
+    }
+
+    public CommentHelper(Element element, Utils utils, List<? extends DocTree> inlineTags) {
+        this.element = element;
+        this.utils = utils;
+        this.inlineTags = inlineTags;
+        this.hasInheritDocTag = utils.hasInlineTag(inlineTags, DocTree.Kind.INHERIT_DOC);
+    }
+
+    /**
+     * Returns true if the method has no comments, or a lone &commat;inheritDoc.
+     *
+     * @return true if there are no comments, false otherwise
+     */
+    public boolean isSimpleOverride() {
+        return inlineTags.isEmpty() ||
+                (inlineTags.size() == 1 && hasInheritDocTag);
+    }
+
+    public boolean hasInheritDocTag(){
+        return this.hasInheritDocTag;
+    }
+
+    public CommentHelper copy() {
+        if (this.element == null) {
+            throw new NullPointerException();
+        }
+        CommentHelper clone = new CommentHelper(this.element, this.utils);
+        return clone;
+    }
+
+    public CommentHelper inherit(CommentHelper chInheritFrom) {
+        List<? extends DocTree> mergedTags = new ArrayList<>();
+
+        if (this.isSimpleOverride())
+            mergedTags = chInheritFrom.inlineTags;
+        else {
+            mergedTags = inheritInlineTags(this, chInheritFrom);
+        }
+
+        return new CommentHelper(this.element, this.utils, mergedTags);
+    }
+
+    List<? extends DocTree> inheritInlineTags(CommentHelper origin, CommentHelper chInheritFrom) {
+        List<DocTree> mergedTags = new ArrayList<>();
+        if (!origin.isSimpleOverride() && !origin.hasInheritDocTag) {

```

Comment:
doesn't `isSimpleOverride` already call `hasInheritDocTag`? Are there scenarios where `isSimpleOverride()` would return false, but `hasInheritDocTag` would return true?

you're more familiar with this codebase/logic so this very well may be the case, but it _seems_ like a candidate for simplification of logic

Reply:
There are two cases we need to do copy comments

Case 1 - The method has no comment  or have only "{@inheritedDoc}". For these methods we just copy everything from latest overridden item or implemented item.

```
    @Override
    public String getName() {};

    /**
     * {@inheritDoc}
     */
    @Override
    public String getVersion() {};
````

Case 2 - The method has comment **and**  "{@inheritedDoc}". For these methods we just copy comments and merge with it's own comment.

```
    /**
     * Get version from an item
     * {@inheritDoc}
     */
    @Override
    public String getVersion() {};
````